core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -229,9 +230,11 @@ private[spark] class Executor(
       // ClosedByInterruptException during execBackend.statusUpdate which causes
       // Executor to crash
       Thread.interrupted()
+      notifyAll()
     }
 
     override def run(): Unit = {
+      Thread.currentThread().setName(threadName)
```

Comment:
Task ids should be unique so therefore this thread name should be unique. Hence, I don't think it's super important to reset the thread's name when returning it to this task thread pool because the thread will just be renamed as soon as it's recycled for a new task and if the task has already exited then it'll be clear from the thread state / context that this is just a completed task's thread that's been returned to the pool.